### PR TITLE
Empty folders and collections

### PR DESCRIPTION
**Pull request by chStaiger**
- Creates the whole subtree when a folder is uploaded or a collection is downloaded; including empty collections and folders respectively.

### CODE REVIEW

File: ibridges/data_operations.py [hunk 1]
```diff
@@ -274,10 +287,19 @@ def _create_local_dest(session: Session, irods_path: IrodsPath, local_path: Path
...
+    # create all folders from collections including empty ones
+    folders = [local_path.joinpath(coll.name, *sc.relative_to(irods_path).parts)
+               for sc in subcolls]
+    for folder in folders:
+        folder.mkdir(parents=True, exist_ok=True)
 
 def _download_collection(session: Session, irods_path: Union[str, IrodsPath], local_path: Path,
                          overwrite: bool = False, ignore_err: bool = False, resc_name: str = '',
-                         options: Optional[dict] = None):
+                         copy_empty_folders: bool = False, options: Optional[dict] = None):
```
Comment (qubixes) [inline ibridges/data_operations.py:302]: Is there a reason why the default value is False here and True in the `download` function? I would suggest putting everything at True by default if not.

File: ibridges/data_operations.py [hunk 2]
```diff
@@ -474,6 +498,16 @@ def _get_data_objects(session: Session,
 
     return objs
 
+def _get_subcoll_paths(session: Session,
+                     coll: irods.collection.iRODSCollection) -> list:
+    """
+    Retrieves all sub collections in a sub tree starting at coll and returns ther IrodsPaths.
```
Comment (qubixes) [inline ibridges/data_operations.py:504]:
```suggestion
    Retrieves all sub collections in a sub tree starting at coll and returns their IrodsPaths.
```

File: ibridges/data_operations.py [hunk 3]
```diff
@@ -474,6 +498,16 @@ def _get_data_objects(session: Session,
 
     return objs
 
+def _get_subcoll_paths(session: Session,
+                     coll: irods.collection.iRODSCollection) -> list:
```
Comment (qubixes) [inline ibridges/data_operations.py:502]: This doesn't look properly lined up, but it could be GitHub.

**Review by qubixes — DISMISSED** [review]
Nice! I think it will lead to less confusion, and if we find out in the end that everyone prefers to have no empty folders/collections, we can just swap the defaults!

File: ibridges/data_operations.py [hunk 4]
```diff
@@ -198,6 +198,9 @@ def _obj_get(session: Session, irods_path: Union[str, IrodsPath], local_path: Un
         options[kw.FORCE_FLAG_KW] = ''
     if resc_name not in ['', None]:
         options[kw.RESC_NAME_KW] = resc_name
+    #Quick fix for #126
+    if Path(local_path).is_dir():
+        local_path = Path(local_path).joinpath(irods_path.parts[-1])
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridges/data_operations.py:203]: Git tells me that there are no merge conflicts ... We will see ;)

File: ibridges/data_operations.py (hunk 1 above)
Comment (chStaiger) [reply ibridges/data_operations.py:302]: Good catch ... yes, the default values should be the same and all default to `True`

File: ibridges/data_operations.py (hunk 3 above)
Comment (chStaiger) [reply ibridges/data_operations.py:502]: It just looks like that in github, on the terminal it looks ok and pylint does not complain.

**Pull request merged**